### PR TITLE
feat(mux): Improved Neovim process detection for Kitty mux integration

### PR DESCRIPTION
**Pull request by nanmoumou**
#147 ability to optimize environmental recognition

### CODE REVIEW

File: kitty/pass_keys.py [hunk 1]
```diff
@@ -1,14 +1,8 @@
...
-from kitty.key_encoding import KeyEvent, parse_shortcut
 
+from kitty.key_encoding import KeyEvent, parse_shortcut
 
-def is_window_vim(window, vim_id):
-    from kittens.tui.loop import debug
-    debug(vim_id)
-    fp = window.child.foreground_processes
-    return any(re.search(vim_id, p['cmdline'][0] if len(p['cmdline']) else '', re.I) for p in fp)
+# from kitty.keys import keyboard_mode_name
```
Comment (mrjones2014) [inline kitty/pass_keys.py:5]: remove commented out line

File: kitty/pass_keys.py [hunk 2]
```diff
@@ -26,58 +20,61 @@ def encode_key_mapping(window, key_mapping):
 
     return window.encoded_key(event)
 
+
```
Comment (mrjones2014) [inline kitty/pass_keys.py:23]: unneeded newline

File: kitty/pass_keys.py [hunk 3]
```diff
@@ -26,58 +20,61 @@ def encode_key_mapping(window, key_mapping):
...
     # only has top neighbor
-    elif direction == 'down' and top_neighbors:
-        boss.active_tab.resize_window('shorter', amount)
+    elif direction == "down" and top_neighbors:
+        boss.active_tab.resize_window("shorter", amount)
     # only has bottom neighbor
-    elif direction == 'down' and bottom_neighbors:
-        boss.active_tab.resize_window('taller', amount)
+    elif direction == "down" and bottom_neighbors:
+        boss.active_tab.resize_window("taller", amount)
```
Comment (mrjones2014) [inline kitty/pass_keys.py:77]: looks like these lines came from different formatter settings. please revert for a clean diff.

File: kitty/pass_keys.py [hunk 4]
```diff
@@ -89,18 +86,21 @@ def handle_result(args, result, target_window_id, boss):
...
 
     if window is None:
         return
-    if is_window_vim(window, vim_id):
+
+    cmd = window.child.foreground_cmdline[0]
+
+    # keyboard_mode_name(window.screen) == "kitty"  You can get kitty.
+    # result: zsh, tmux, nvim
+    if cmd[-4:] == "nvim":
```
Comment (mrjones2014) [inline kitty/pass_keys.py:99]: correct me if I'm wrong but I don't think this works if, for example, you run `nvim ./some/file/name.txt` or if you run Neovim via a `vim` shell alias.

File: kitty/pass_keys.py [hunk 5]
```diff
@@ -89,18 +86,21 @@ def handle_result(args, result, target_window_id, boss):
...
         for keymap in key_mapping.split(">"):
             encoded = encode_key_mapping(window, keymap)
             window.write_to_child(encoded)
-    elif action == 'neighboring_window':
+    elif action == "neighboring_window":
         boss.active_tab.neighboring_window(direction)
-    elif action == 'relative_resize':
+    elif action == "relative_resize":
```
Comment (mrjones2014) [inline kitty/pass_keys.py:105]: please revert formatting changes for a clean diff.

**Review by mrjones2014 — CHANGES_REQUESTED** [review]

**Comment by mrjones2014** [comment]
Hi @nanmoumou just wondering if you're planning to update the PR

**Comment by nanmoumou** [comment]
> 你好@nanmoumou只是想知道您是否打算更新 PR

Give it up for now.
I've found a small issue where I can't seem to get the correct results when using pipenv, with or without regular expressions.

**Pull request closed without merging**